src/cgame/cg_rocket_datasource.cpp

```diff
@@ -34,6 +34,47 @@ Maryland 20850 USA.
 
 #include "cg_local.h"
 
+static void StripNonPrintingCharacters( std::string& str )
+{
+	str.erase(
+		std::remove_if(
+			str.begin(),
+			str.end(),
+			[]( unsigned char c ){
+				return !std::isprint( c );
+			}),
+		str.end() );
+}
+
+static void StripSpacingCharacters( std::string& str )
+{
+	str.erase(
+		std::remove_if(
+			str.begin(),
+			str.end(),
+			[]( unsigned char c ){
+				return std::isspace( c );
+			}),
+		str.end() );
+}
+
+static std::string SanitizeString( std::string original, std::string fallback )
+{
+	std::string printable = original;
+	StripNonPrintingCharacters( printable );
+
+	std::string stripped = printable;
+	Color::StripColors( stripped );
```

Comment:
You have to use the return value of StripColors

Reply:
Something is probably wrong in that `StripColors` code, I first did that:

```
std:string stripped = Color::StripColors( original );
```

But then all servers were color-stripped.

It looks like `Color::StripColors` modifies the string in place and the return is useless.

If that's not expected, we have a bug.

Reply:
Well, it looks like it was a bug. Too bad I didn't kept the code triggering it (earlier version of this PR), as it was the exact opposite (and with GCC) of:

- https://github.com/DaemonEngine/Daemon/issues/839

The gcc bug I got with an early version of the commit was: input and ouput are both stripped.
The clang bug being: neither input and output are stripped.

Reply:
Ah no I get it, my earlier code was doing:

```
std::string stripped = Color::StripColors( node->name );
```

But `node->name` is a `char*` and then the function modifies the input in place…

We have a very bad api on that part.

This is the function:

```

char *StripColors( char *string )
{
	char *out = string;

	for ( const auto& token : Parser( string ) )
	{
		Str::StringView text = token.PlainText();
		memmove( out, text.begin(), text.size() );
		out += text.size();
	}

	*out = '\0';
	return string;
}
```

GCC modifies both strings, Clang modifies none.

Reply:
Fixed.